File: bitget-php-sdk-api/src/api/v2/SpotWalletApi.php
<?php


namespace bitget\api\v2;


use bitget\internal\BitgetApiClient;

class SpotWalletApi
{
    var BitgetApiClient $BitgetApiClient;

    public function __construct($BitgetApiClient)
    {
        $this->BitgetApiClient = $BitgetApiClient;
    }

    public function transfer($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/spot/wallet/transfer", $params);
    }

    public function depositAddress($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/spot/wallet/deposit-address", $params);
    }

    public function withdrawal($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/spot/wallet/withdrawal", $params);
    }

    public function withdrawalRecords($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/spot/wallet/withdrawal-records", $params);
    }

    public function depositRecords($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/spot/wallet/deposit-records", $params);
    }

}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v2/SpotWalletApi.java
package com.bitget.openapi.api.v2;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface SpotWalletApi {

    @POST("/api/v2/spot/wallet/transfer")
    Call<ResponseResult> transfer(@Body Map<String, String> paramMap);

    @GET("/api/v2/spot/wallet/deposit-address")
    Call<ResponseResult> depositAddress(@QueryMap Map<String, String> paramMap);

    @POST("/api/v2/spot/wallet/withdrawal")
    Call<ResponseResult> withdrawal(@Body Map<String, String> paramMap);

    @GET("/api/v2/spot/wallet/withdrawal-records")
    Call<ResponseResult> withdrawalRecords(@QueryMap Map<String, String> paramMap);

    @GET("/api/v2/spot/wallet/deposit-records")
    Call<ResponseResult> depositRecords(@QueryMap Map<String, String> paramMap);
}
